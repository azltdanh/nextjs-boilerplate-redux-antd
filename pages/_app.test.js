import App from './_app';
import PageNotFound from './404';

// mock router history
const mockHistoryPush = jest.fn();
jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    pathname: '/',
    push: mockHistoryPush,
  }),
}));

describe('App', () => {
  it('renders', () => {
    const props = {
      Component: PageNotFound,
      pageProps: {},
    };
    const wrapper = mount(<App {...props} />);
    expect(wrapper.find(PageNotFound)).toMatchSnapshot();
  });
});
