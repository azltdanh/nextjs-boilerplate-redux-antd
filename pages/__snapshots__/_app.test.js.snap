// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<PageNotFound>
  <h1>
    404 - Page Not Found
  </h1>
</PageNotFound>
`;
